src/launchbox/exchange/LaunchboxExchange.sol

```diff
@@ -118,13 +124,82 @@ contract LaunchboxExchange {
     function getAmountOutWithFee(uint256 amountIn, uint256 reserveIn, uint256 reserveOut, uint256 _tradeFee)
         internal
         pure
-        returns (uint256, uint256)
+        returns (uint256 amountOut, uint256 fee)
     {
         require(amountIn > 0, "Amount in must be greater than 0");
-        uint256 amountInWithFee = amountIn * (1000 - _tradeFee);
-        uint256 numerator = amountInWithFee * reserveOut;
-        uint256 denominator = (reserveIn * 1000) + amountInWithFee;
-        return (numerator / denominator, (amountIn * _tradeFee) / 1000);
+        require(_tradeFee <= FEE_DENOMINATOR, "Trade fee must be less than or equal to 100%");
+        require(reserveIn > 0 && reserveOut > 0, "Reserves must be greater than 0");
+
+        // Calculate fee
+        fee = mulDiv(amountIn, _tradeFee, FEE_DENOMINATOR);
+
+        // Calculate amountInWithFee
+        uint256 amountInWithFee = amountIn - fee;
+
+        // Calculate numerator and denominator
+        uint256 numerator = mulDiv(amountInWithFee, reserveOut, 1);
+        uint256 denominator = reserveIn * FEE_DENOMINATOR + amountInWithFee;
```

Comment:
This changes the Math.

earlier the formula was: 

```math
\frac{\text{amountIn} \times (\text{FEE\_DENOMINATOR} - \text{\_tradeFee}) \times \text{reserveOut}}{(\text{reserveIn} \times \text{FEE\_DENOMINATOR}) + (\text{amountIn} \times (\text{FEE\_DENOMINATOR} - \text{\_tradeFee}))}
```

But now it is:
```math
\frac{\left( \text{amountIn} \times \left( \text{FEE\_DENOMINATOR} - \text{\_tradeFee} \right) / \text{FEE\_DENOMINATOR} \right) \times \text{reserveOut}}{\left( \text{reserveIn} \times \text{FEE\_DENOMINATOR} \right) + \left( \text{amountIn} \times \left( \text{FEE\_DENOMINATOR} - \text{\_tradeFee} \right) \right)}
```

Reply:
let me update it

Reply:
done